src/source/tile.js

```diff
@@ -434,7 +435,8 @@ class Tile {
 
             bucket.update(sourceLayerStates, sourceLayer);
             if (painter && painter.style) {
-                this.queryPadding = Math.max(this.queryPadding, painter.style.getLayer(bucket.layerIds[0]).queryRadius(bucket));
+                const index = bucket.layerIds.indexOf(id);
+                this.queryPadding = Math.max(this.queryPadding, painter.style.getLayer(bucket.layerIds[index]).queryRadius(bucket));
```

Comment:
`layerIds[layerIds.indexOf(id)]` is simply `id`, so no need to do `indexOf` lookups in both places.